src/test/java/seedu/address/model/ModelManagerTest.java

```diff
@@ -216,6 +226,85 @@ public void getFilteredModuleList_modifyList_throwsUnsupportedOperationException
         assertThrows(UnsupportedOperationException.class, () -> modelManager.getFilteredModuleList().remove(0));
     }
 
+    @Test
+    public void hasPersonInFilteredList_nullPerson_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> modelManager.hasPersonInFilteredList(null));
+    }
+
+    @Test
+    public void hasPersonInFilteredList_personNotInAddressBook_returnsFalse() {
+        assertFalse(modelManager.hasPersonInFilteredList(AMY));
+    }
+
+    @Test
+    public void hasPersonInFilteredList_personInAddressBookAndFilteredList_returnsTrue() {
+        modelManager.addPerson(AMY);
+        modelManager.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        assertTrue(modelManager.hasPersonInFilteredList(AMY));
+    }
+
+    @Test
+    public void hasPersonInFilteredList_personInAddressBookButNotInFilteredList_returnsFalse() {
+        modelManager.addPerson(AMY);
+        modelManager.updateFilteredPersonList(PREDICATE_SHOW_ZERO_PERSON);
+        assertFalse(modelManager.hasPersonInFilteredList(AMY));
+    }
+
+    @Test
+    public void hasPersonInFilteredList_personWithSameNameButDifferentFieldsInFilteredList_returnsTrue() {
+        Person personWithSameNameButDifferentFields =
+                new PersonBuilder().withName(VALID_NAME_AMY)
+                        .withEmail("test@gmail.com")
```

Comment:
```suggestion
                        .withEmail(VALID_EMAIL_BOB)
```
Might it be better to use Bob's fields for this test? Would help us to reduce the usage of [magic literals](https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers) in our test cases.

Reply:
Good point! I shall adopt your suggestion 💯